application/common/models/User.php

```diff
@@ -1115,4 +1111,34 @@ public static function numberWithOneMfaNotBackupCodes()
         ")->queryOne();
         return $result['COUNT(*)'];
     }
+
+    protected function isExpired(): bool
+    {
+        return $this->expires_on != null && MySqlDateTime::isBefore($this->expires_on, time());
+    }
+
+    protected function deactivate(): void
```

Comment:
It may be worth clarifying some details about this method:

* Based on (new) line 1127, it looks like this is only intended to be used for deactivating an _expired_ user. If that's truly the only intended usage, what would you think about changing the name from `deactivate()` to `deactivateExpiredUser()`?
* I would normally expect a method to do what it says or, if it can't, then throw an exception. Given that this one simply logs any errors and returns normally (which is understandable, given when it's called), it may be worth documenting that behavior in a DocBlock above the method.

Reply:
Good thoughts. Updated.